Broken Tags test on Travis

Travis is failing with the following error (see, for example, job [346239552](https://www.travis-ci.org/agda/agda/jobs/346239552)):

```
======================================================================
================================ TAGS ================================
======================================================================
make -C ./src/full TAGS
make[1]: Entering directory `/home/travis/build/agda/agda/src/full'
make -C ../hTags
make[2]: Entering directory `/home/travis/build/agda/agda/src/hTags'
cabal install --only-dependencies
Resolving dependencies...
In order, the following would be installed:
mtl-2.2.2 (reinstall) (changes: transformers-0.5.5.0 -> 0.4.2.0)
cabal: The following packages are likely to be broken by the reinstalls:
...
Use --force-reinstalls if you want to install anyway.
make[2]: *** [dist/setup-config] Error 1
```

--- a/Agda.cabal
+++ b/Agda.cabal
@@ -162,16 +162,20 @@ library
                 , template-haskell >= 2.10.0.0 && < 2.14
                 , text >= 0.11.3.1 && < 1.3
                 , time >= 1.5.0.1 && < 1.9
-                -- In hTags the mtl library must be compiled with the
-                -- version of transformers shipped with GHC. The lower
-                -- bound is the version of transformers shipped with
-                -- GHC 7.10.3 (it is not necessary set the lower bound
-                -- to the version shipped with GHC 7.8.4 because in
-                -- this case Agda and hTags use a different version of
-                -- mtl).
-                , transformers >= 0.4.2.0 && < 0.6
                 , unordered-containers >= 0.2.5.0 && < 0.3
                 , uri-encode >= 1.5.0.4 && < 1.6
+
+-- In hTags the mtl library must be compiled with the version of
+-- transformers shipped with GHC, so we use that version in Agda (see,
+-- for example, Issue 2983).
+  if impl(ghc >= 8.4)
+    build-depends: transformers == 0.5.5.0
+
+  if impl(ghc >= 8.0) && impl(ghc < 8.4)
+    build-depends: transformers == 0.5.2.0
+
+  if impl(ghc >= 7.10.3) && impl(ghc < 8.0)
+    build-depends: transformers == 0.4.2.0
 
   -- zlib >= 0.6.1 is broken with GHC < 7.10.3. See Issue 1518.
   if impl(ghc < 7.10.3)

